ethcontract-common/src/artifact.rs

```diff
@@ -99,3 +124,51 @@ impl Default for Artifact {
         Artifact::new()
     }
 }
+
+/// Result of inserting a nre contract into an artifact.
+pub struct InsertResult<'a> {
+    /// Reference to the newly inserted contract.
+    pub inserted_contract: ContractMut<'a>,
+
+    /// If insert operation replaced an old contract, it will appear here.
+    pub old_contract: Option<Contract>,
+}
+
+/// A wrapper that allows mutating contract
+/// but doesn't allow changing its name.
+pub struct ContractMut<'a>(&'a mut Contract);
+
+impl<'a> ContractMut<'a> {
+    /// Get mutable access to abi.
+    pub fn abi_mut(&mut self) -> &mut Abi {
+        &mut self.0.abi
+    }
+
+    /// Get mutable access to bytecode.
+    pub fn bytecode_mut(&mut self) -> &mut Bytecode {
+        &mut self.0.bytecode
+    }
+
+    /// Get mutable access to networks.
+    pub fn networks_mut(&mut self) -> &mut HashMap<String, Network> {
+        &mut self.0.networks
+    }
+
+    /// Get mutable access to devdoc.
+    pub fn devdoc_mut(&mut self) -> &mut Documentation {
```

Comment:
One thing I was thinking - since the documentation in the contract is only really needed by the code generation, do you think it would be possible to separate it from `Contract` (so that it only contains things needed by the runtime) and fit it into the `Artifact` somehow.

Just rambling a little - haven't really thought this suggestion through.

Reply:
We could wrap `Contract` into `ContractWithDocs`, but do we really need to? I mean, if we don't need those fields, we can just keep them empty? Besides, what if someone actually needs them, i.e. for UI purposes?

Reply:
I think you're right, lets leave it as is then.